File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/quests/instances/FromHell.java
package it.paranoidsquirrels.idleguildmaster.storage.data.quests.instances;

import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest;
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.QuestsManager;

/* JADX INFO: loaded from: classes3.dex */
public class FromHell extends Quest {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    public void calculateTargetProgress(int i) {
        this.targetProgress = (((long) (i - 10)) * 5) + 15;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    protected void configure() {
        this.idName = R.string.quest_from_hell_name;
        this.idDescription = R.string.quest_from_hell_description;
        this.defaultRarity = 2;
        this.minimumDifficulty = 10;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    public void realignStaticReference() {
        QuestsManager.fromHell = this;
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/quests/instances/FromHell.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.quests.instances

import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.QuestsManager

class FromHell : Quest() {
    override fun calculateTargetProgress(difficulty: Int) {
        targetProgress = (((difficulty - 10).toLong() * 5) + 15).toLong()
    }

    override fun configure() {
        idName = R.string.quest_from_hell_name
        idDescription = R.string.quest_from_hell_description
        defaultRarity = 2
        minimumDifficulty = 10
    }

    override fun realignStaticReference() {
        QuestsManager.fromHell = this
    }
}
